pandas/core/arrays/string_.py

```diff
@@ -0,0 +1,217 @@
+from typing import Type
+
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.dtypes.common import pandas_dtype
+from pandas.core.dtypes.dtypes import register_extension_dtype
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+from pandas.core.dtypes.inference import is_array_like
+
+from pandas.core.arrays import PandasArray
+from pandas.core.construction import extract_array
+
+
+@register_extension_dtype
+class StringDtype(ExtensionDtype):
+    """
+    Extension dtype for text data.
+
+    .. versionadded:: 1.0.0
+
+    Attributes
+    ----------
+    None
+
+    Methods
+    -------
+    None
+
+    Examples
+    --------
+    >>> pd.StringDtype()
+    StringDtype
+    """
+
+    @property
+    def na_value(self):
+        """
+        StringDtype uses :attr:`numpy.nan` as the missing NA value.
+        """
+        return np.nan
+
+    @property
+    def type(self) -> Type:
+        return str
+
+    @property
+    def name(self) -> str:
+        """
+        The alias for StringDtype is ``'string'``.
+        """
+        return "string"
+
+    @classmethod
+    def construct_from_string(cls, string: str):
```

Comment:
I think can annotate this return with `ExtensionDtype`

Reply:
OK. Technically, this is just `-> StringDtype`. We only use `super()` for error checking. But mypy doesn't like `StringDtype` since the return type of the parent is `ExtensionDtype`.